pkg/execution/executor/executor.go

```diff
@@ -648,14 +649,16 @@ func (e *executor) Execute(ctx context.Context, id state.Identifier, item queue.
 			attribute.String(consts.OtelSysIdempotencyKey, id.IdempotencyKey()),
 		),
 	)
-	if len(id.EventIDs) > 1 {
+	if id.BatchID != nil {
 		fnSpan.SetAttributes(attribute.String(consts.OtelSysBatchID, id.BatchID.String()))
 	}
 	for _, e := range s.Events() {
 		if byt, err := json.Marshal(e); err == nil {
-			fnSpan.AddEvent(string(byt), trace.WithAttributes(
-				attribute.Bool(consts.OtelSysEventData, true),
-			))
+			if b, err := jcs.Transform(byt); err == nil {
```

Comment:
And similarly, here, why are we transforming, too?  We've also already transformed on the Schedule call, so this seems to be extra work, CPU, latency on every step?

Reply:
removing this as discussed